File: 1304-find-n-unique-integers-sum-up-to-zero/1304-find-n-unique-integers-sum-up-to-zero.py
class Solution:
    def sumZero(self, n: int) -> List[int]:
        ans = [0] * n
        idx, num = 0, 1
        
        # for odd length, we need extra 0 digit
        if (n & 1) == 1: idx += 1
        
        while idx < n:
            ans[idx] = -num
            ans[idx+1] = num
            idx += 2
            num += 1

        return ans

File: 1304-find-n-unique-integers-sum-up-to-zero/1304-find-n-unique-integers-sum-up-to-zero.cs
public class Solution {
    public int[] SumZero(int n) {
        var ans = new int[n];
        int idx = 0, num = 1;
        
        // for odd length, we need extra digit
        if((n & 1) == 1)
            ans[idx++] = 0;
        
        while(idx < n){
            ans[idx++] = -num;
            ans[idx++] = num++;
        }

        return ans;
    }
}